tree-sitter-stack-graphs/src/cli/index.rs

```diff
@@ -86,57 +92,69 @@ impl AnalyzeArgs {
         }
     }
 
-    pub fn run(&self, loader: &mut Loader) -> anyhow::Result<()> {
+    pub fn run(&self, db_path: &Path, loader: &mut Loader) -> anyhow::Result<()> {
         if self.wait_at_start {
-            self.wait_for_input()?;
+            wait_for_input()?;
         }
         let mut seen_mark = false;
+        let mut db = SQLiteWriter::open(&db_path)?;
         for source_path in &self.source_paths {
+            let source_path = source_path.canonicalize()?;
             if source_path.is_dir() {
-                let source_root = source_path;
+                let source_root = &source_path;
                 for source_entry in WalkDir::new(source_root)
                     .follow_links(true)
                     .sort_by_file_name()
                     .into_iter()
                     .filter_map(|e| e.ok())
                     .filter(|e| e.file_type().is_file())
                 {
-                    let source_path = source_entry.path();
-                    self.analyze_file(source_root, source_path, loader, &mut seen_mark)?;
+                    let source_path = source_entry.path().canonicalize()?;
+                    self.analyze_file(
+                        source_root,
+                        &source_path,
```

Comment:
Is it intended that only one of these is a reference? Same for lines 128-129

Reply:
Good question! Both were passed by reference, but `source_root` was already a reference, while `source_path` was a value. However, Rick made a nice suggestion above that allows using `source_path` now.